spring-pulsar/src/main/java/org/springframework/pulsar/core/PulsarTemplate.java

```diff
@@ -42,13 +44,25 @@
 
 	private final PulsarProducerFactory<T> producerFactory;
 
+	private final List<ProducerInterceptor> interceptors;
+
 	/**
 	 * Constructs a template instance.
 	 * @param producerFactory the producer factory used to create the backing Pulsar
 	 * producers.
 	 */
 	public PulsarTemplate(PulsarProducerFactory<T> producerFactory) {
+		this(producerFactory, null);
+	}
+
+	/**
```

Comment:
> Since producer interceptors are optional, what do you think about providing a mutator on PulsarTemplate rather than setting it through the constructor? I say that because if we go with the constructor approach, we may end up with more constructors for each optional concern like this in the future. But, let's discuss; I am open either way.

@sobychacko I understand wanting to avoid `<init>` bloat as it can get out of hand quick. In this case I am favor of keeping it as-is because 
1. it is a natural fit w/ the use of [`ObjectProvider`](https://github.com/spring-projects-experimental/spring-pulsar/pull/59#discussion_r942860574) in the auto-config
2. Similar to constructor bloat is mix of constructor and `setXYZ` (aka setter bloat). I think the proper way to avoid both constructor and setter bloat would be to introduce a builder pattern. However, I think we do not need to worry about that right now. 

Wdyt?

Reply:
Yep, let's keep it that way. We will see how the template evolves.